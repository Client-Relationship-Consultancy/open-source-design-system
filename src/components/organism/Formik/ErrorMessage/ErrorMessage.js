import React from "react"
import { ErrorMessage as Error } from "formik"
import PropTypes from "prop-types"
import styled from "styled-components"
import { colourPalette } from "brandColours"

export const StyledErrorMessage = styled.div`
  font-weight: bold;
  color: ${props => props.theme.error.main.hex};
  margin-top: 0.5rem;
`
StyledErrorMessage.defaultProps = {
  theme: colourPalette.examplePalette,
}
StyledErrorMessage.displayName = "StyledErrorMessage"

const ErrorMessage = props => {
  const { name, id, className } = props
  return <Error name={name} component={StyledErrorMessage} id={id} className={className} />
}
ErrorMessage.displayName = "ErrorMessage"

ErrorMessage.propTypes = {
  name: PropTypes.string,
  id: PropTypes.string,
  className: PropTypes.string,
}

export default ErrorMessage
