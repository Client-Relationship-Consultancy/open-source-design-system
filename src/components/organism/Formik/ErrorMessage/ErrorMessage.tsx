import React from "react";
import styled from "styled-components";
import { colourPalette } from "../../../../brandColours";

export const StyledErrorMessage = styled.div`
  font-weight: bold;
  color: ${(props) => props.theme.error.main.hex};
  margin-top: 0.5rem;
`;
StyledErrorMessage.defaultProps = {
  theme: colourPalette.examplePalette,
};
StyledErrorMessage.displayName = "StyledErrorMessage";

export const StyledHelperCaption = styled.div`
  color: ${(props) => props.theme.secondary.main.hex};
  margin-top: 0.5rem;
`;
StyledHelperCaption.defaultProps = {
  theme: colourPalette.examplePalette,
};
StyledHelperCaption.displayName = "StyledHelperCaption";

interface IProps {
  error: string;
  id?: string;
  className: string;
  caption?: string;
  isError: boolean;
}

export class ErrorMessage extends React.PureComponent<IProps> {
  render() {
    const { error, id, className, caption, isError } = this.props;
    const errorClassName = className ? `ErrorMessage ${className}` : "ErrorMessage";
    return (
      <>
        {isError ? (
          <StyledErrorMessage id={id} className={errorClassName}>
            {error}
          </StyledErrorMessage>
        ) : (
          caption && (
            <StyledHelperCaption id={id} className={errorClassName}>
              {caption}
            </StyledHelperCaption>
          )
        )}
      </>
    );
  }
}

export default ErrorMessage;
