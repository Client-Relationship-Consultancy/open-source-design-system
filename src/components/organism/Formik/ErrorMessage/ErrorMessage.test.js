import React from "react"
import { mount } from "enzyme"
import toJson from "enzyme-to-json"
import ErrorMessage, { StyledErrorMessage } from "./ErrorMessage"
import "jest-styled-components"

describe("ErrorMessage Component Test", () => {
  const component = mount(<ErrorMessage name="email" />)
  const componentStyled = mount(<StyledErrorMessage name="email" />)

  it("ErrorMessage sould match last snapshot", () => {
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it("StyledErrorMessage should match last snapshot", () => {
    const tree = toJson(componentStyled)
    expect(tree).toMatchSnapshot()
  })
})
