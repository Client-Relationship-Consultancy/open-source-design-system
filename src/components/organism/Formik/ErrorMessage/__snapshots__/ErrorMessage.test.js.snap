// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Component Test ErrorMessage sould match last snapshot 1`] = `
<ErrorMessage
  name="email"
>
  <FormikConnect(ErrorMessageImpl)
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "_foldedDefaultProps": Object {
          "theme": Object {
            "action": Object {
              "dark": Object {
                "RGB": "0,47,108",
                "hex": "#002F6C",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "98,194,248",
                "hex": "#62C2F8",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "1,87,155",
                "hex": "#01579B",
                "on": "#FFFFFF",
              },
            },
            "background": Object {
              "RGB": "246,246,246",
              "hex": "#F6F6F6",
            },
            "black": Object {
              "main": Object {
                "RGB": "39,39,39",
                "hex": "#272727",
                "on": "#FFFFFF",
              },
              "tint20": Object {
                "RGB": "236,236,236",
                "hex": "#ECECEC",
                "on": "#272727",
              },
              "tint40": Object {
                "RGB": "199,199,199",
                "hex": "#C7C7C7",
                "on": "#272727",
              },
              "tint60": Object {
                "RGB": "160,160,160",
                "hex": "#A0A0A0",
                "on": "#FFFFFF",
              },
              "tint80": Object {
                "RGB": "105,105,105",
                "hex": "#696969",
                "on": "#FFFFFF",
              },
            },
            "complimentary": Object {
              "dark": Object {
                "RGB": "250,175,64",
                "hex": "#FAAF40",
                "on": "#272727",
              },
              "light": Object {
                "RGB": "249,240,21",
                "hex": "#F9F015",
                "on": "#272727",
              },
              "main": Object {
                "RGB": "253,216,53",
                "hex": "#FDD835",
                "on": "#272727",
              },
            },
            "error": Object {
              "dark": Object {
                "RGB": "143,6,6",
                "hex": "#8F0606",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "220,16,16",
                "hex": "#DC1010",
                "on": "#FFFFFF",
              },
            },
            "primary": Object {
              "dark": Object {
                "RGB": "11,102,54",
                "hex": "#0B6636",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "145,208,62",
                "hex": "#91D03E",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "1,145,68",
                "hex": "#019144",
                "on": "#FFFFFF",
              },
            },
            "secondary": Object {
              "dark": Object {
                "RGB": "72,140,155",
                "hex": "#488C9B",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "162,235,252",
                "hex": "#A2EBFC",
                "on": "#272727",
              },
              "main": Object {
                "RGB": "118,182,196",
                "hex": "#76B6C4",
                "on": "#FFFFFF",
              },
            },
            "surface": Object {
              "RGB": "255,255,255",
              "hex": "#FFFFFF",
              "on": "#272727",
            },
            "white": Object {
              "RGB": "255,255,255",
              "hex": "#FFFFFF",
              "on": "#272727",
            },
          },
        },
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "ErrorMessage__StyledErrorMessage-sc-1d8clug-0",
          "isStatic": false,
          "lastClassName": "fnPqyU",
          "rules": Array [
            "font-weight:bold;color:",
            [Function],
            ";margin-top:0.5rem;",
          ],
        },
        "displayName": "StyledErrorMessage",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "ErrorMessage__StyledErrorMessage-sc-1d8clug-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    name="email"
  >
    <ErrorMessageImpl
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "_foldedDefaultProps": Object {
            "theme": Object {
              "action": Object {
                "dark": Object {
                  "RGB": "0,47,108",
                  "hex": "#002F6C",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "98,194,248",
                  "hex": "#62C2F8",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "1,87,155",
                  "hex": "#01579B",
                  "on": "#FFFFFF",
                },
              },
              "background": Object {
                "RGB": "246,246,246",
                "hex": "#F6F6F6",
              },
              "black": Object {
                "main": Object {
                  "RGB": "39,39,39",
                  "hex": "#272727",
                  "on": "#FFFFFF",
                },
                "tint20": Object {
                  "RGB": "236,236,236",
                  "hex": "#ECECEC",
                  "on": "#272727",
                },
                "tint40": Object {
                  "RGB": "199,199,199",
                  "hex": "#C7C7C7",
                  "on": "#272727",
                },
                "tint60": Object {
                  "RGB": "160,160,160",
                  "hex": "#A0A0A0",
                  "on": "#FFFFFF",
                },
                "tint80": Object {
                  "RGB": "105,105,105",
                  "hex": "#696969",
                  "on": "#FFFFFF",
                },
              },
              "complimentary": Object {
                "dark": Object {
                  "RGB": "250,175,64",
                  "hex": "#FAAF40",
                  "on": "#272727",
                },
                "light": Object {
                  "RGB": "249,240,21",
                  "hex": "#F9F015",
                  "on": "#272727",
                },
                "main": Object {
                  "RGB": "253,216,53",
                  "hex": "#FDD835",
                  "on": "#272727",
                },
              },
              "error": Object {
                "dark": Object {
                  "RGB": "143,6,6",
                  "hex": "#8F0606",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "220,16,16",
                  "hex": "#DC1010",
                  "on": "#FFFFFF",
                },
              },
              "primary": Object {
                "dark": Object {
                  "RGB": "11,102,54",
                  "hex": "#0B6636",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "145,208,62",
                  "hex": "#91D03E",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "1,145,68",
                  "hex": "#019144",
                  "on": "#FFFFFF",
                },
              },
              "secondary": Object {
                "dark": Object {
                  "RGB": "72,140,155",
                  "hex": "#488C9B",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "162,235,252",
                  "hex": "#A2EBFC",
                  "on": "#272727",
                },
                "main": Object {
                  "RGB": "118,182,196",
                  "hex": "#76B6C4",
                  "on": "#FFFFFF",
                },
              },
              "surface": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
              "white": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
            },
          },
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "ErrorMessage__StyledErrorMessage-sc-1d8clug-0",
            "isStatic": false,
            "lastClassName": "fnPqyU",
            "rules": Array [
              "font-weight:bold;color:",
              [Function],
              ";margin-top:0.5rem;",
            ],
          },
          "displayName": "StyledErrorMessage",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "ErrorMessage__StyledErrorMessage-sc-1d8clug-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      formik={Object {}}
      name="email"
    />
  </FormikConnect(ErrorMessageImpl)>
</ErrorMessage>
`;

exports[`ErrorMessage Component Test StyledErrorMessage should match last snapshot 1`] = `
.c0 {
  font-weight: bold;
  color: #DC1010;
  margin-top: 0.5rem;
}

<StyledErrorMessage
  name="email"
  theme={
    Object {
      "action": Object {
        "dark": Object {
          "RGB": "0,47,108",
          "hex": "#002F6C",
          "on": "#FFFFFF",
        },
        "light": Object {
          "RGB": "98,194,248",
          "hex": "#62C2F8",
          "on": "#FFFFFF",
        },
        "main": Object {
          "RGB": "1,87,155",
          "hex": "#01579B",
          "on": "#FFFFFF",
        },
      },
      "background": Object {
        "RGB": "246,246,246",
        "hex": "#F6F6F6",
      },
      "black": Object {
        "main": Object {
          "RGB": "39,39,39",
          "hex": "#272727",
          "on": "#FFFFFF",
        },
        "tint20": Object {
          "RGB": "236,236,236",
          "hex": "#ECECEC",
          "on": "#272727",
        },
        "tint40": Object {
          "RGB": "199,199,199",
          "hex": "#C7C7C7",
          "on": "#272727",
        },
        "tint60": Object {
          "RGB": "160,160,160",
          "hex": "#A0A0A0",
          "on": "#FFFFFF",
        },
        "tint80": Object {
          "RGB": "105,105,105",
          "hex": "#696969",
          "on": "#FFFFFF",
        },
      },
      "complimentary": Object {
        "dark": Object {
          "RGB": "250,175,64",
          "hex": "#FAAF40",
          "on": "#272727",
        },
        "light": Object {
          "RGB": "249,240,21",
          "hex": "#F9F015",
          "on": "#272727",
        },
        "main": Object {
          "RGB": "253,216,53",
          "hex": "#FDD835",
          "on": "#272727",
        },
      },
      "error": Object {
        "dark": Object {
          "RGB": "143,6,6",
          "hex": "#8F0606",
          "on": "#FFFFFF",
        },
        "main": Object {
          "RGB": "220,16,16",
          "hex": "#DC1010",
          "on": "#FFFFFF",
        },
      },
      "primary": Object {
        "dark": Object {
          "RGB": "11,102,54",
          "hex": "#0B6636",
          "on": "#FFFFFF",
        },
        "light": Object {
          "RGB": "145,208,62",
          "hex": "#91D03E",
          "on": "#FFFFFF",
        },
        "main": Object {
          "RGB": "1,145,68",
          "hex": "#019144",
          "on": "#FFFFFF",
        },
      },
      "secondary": Object {
        "dark": Object {
          "RGB": "72,140,155",
          "hex": "#488C9B",
          "on": "#FFFFFF",
        },
        "light": Object {
          "RGB": "162,235,252",
          "hex": "#A2EBFC",
          "on": "#272727",
        },
        "main": Object {
          "RGB": "118,182,196",
          "hex": "#76B6C4",
          "on": "#FFFFFF",
        },
      },
      "surface": Object {
        "RGB": "255,255,255",
        "hex": "#FFFFFF",
        "on": "#272727",
      },
      "white": Object {
        "RGB": "255,255,255",
        "hex": "#FFFFFF",
        "on": "#272727",
      },
    }
  }
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "_foldedDefaultProps": Object {
          "theme": Object {
            "action": Object {
              "dark": Object {
                "RGB": "0,47,108",
                "hex": "#002F6C",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "98,194,248",
                "hex": "#62C2F8",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "1,87,155",
                "hex": "#01579B",
                "on": "#FFFFFF",
              },
            },
            "background": Object {
              "RGB": "246,246,246",
              "hex": "#F6F6F6",
            },
            "black": Object {
              "main": Object {
                "RGB": "39,39,39",
                "hex": "#272727",
                "on": "#FFFFFF",
              },
              "tint20": Object {
                "RGB": "236,236,236",
                "hex": "#ECECEC",
                "on": "#272727",
              },
              "tint40": Object {
                "RGB": "199,199,199",
                "hex": "#C7C7C7",
                "on": "#272727",
              },
              "tint60": Object {
                "RGB": "160,160,160",
                "hex": "#A0A0A0",
                "on": "#FFFFFF",
              },
              "tint80": Object {
                "RGB": "105,105,105",
                "hex": "#696969",
                "on": "#FFFFFF",
              },
            },
            "complimentary": Object {
              "dark": Object {
                "RGB": "250,175,64",
                "hex": "#FAAF40",
                "on": "#272727",
              },
              "light": Object {
                "RGB": "249,240,21",
                "hex": "#F9F015",
                "on": "#272727",
              },
              "main": Object {
                "RGB": "253,216,53",
                "hex": "#FDD835",
                "on": "#272727",
              },
            },
            "error": Object {
              "dark": Object {
                "RGB": "143,6,6",
                "hex": "#8F0606",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "220,16,16",
                "hex": "#DC1010",
                "on": "#FFFFFF",
              },
            },
            "primary": Object {
              "dark": Object {
                "RGB": "11,102,54",
                "hex": "#0B6636",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "145,208,62",
                "hex": "#91D03E",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "1,145,68",
                "hex": "#019144",
                "on": "#FFFFFF",
              },
            },
            "secondary": Object {
              "dark": Object {
                "RGB": "72,140,155",
                "hex": "#488C9B",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "162,235,252",
                "hex": "#A2EBFC",
                "on": "#272727",
              },
              "main": Object {
                "RGB": "118,182,196",
                "hex": "#76B6C4",
                "on": "#FFFFFF",
              },
            },
            "surface": Object {
              "RGB": "255,255,255",
              "hex": "#FFFFFF",
              "on": "#272727",
            },
            "white": Object {
              "RGB": "255,255,255",
              "hex": "#FFFFFF",
              "on": "#272727",
            },
          },
        },
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "ErrorMessage__StyledErrorMessage-sc-1d8clug-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "font-weight:bold;color:",
            [Function],
            ";margin-top:0.5rem;",
          ],
        },
        "displayName": "StyledErrorMessage",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "ErrorMessage__StyledErrorMessage-sc-1d8clug-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    name="email"
    theme={
      Object {
        "action": Object {
          "dark": Object {
            "RGB": "0,47,108",
            "hex": "#002F6C",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "98,194,248",
            "hex": "#62C2F8",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "1,87,155",
            "hex": "#01579B",
            "on": "#FFFFFF",
          },
        },
        "background": Object {
          "RGB": "246,246,246",
          "hex": "#F6F6F6",
        },
        "black": Object {
          "main": Object {
            "RGB": "39,39,39",
            "hex": "#272727",
            "on": "#FFFFFF",
          },
          "tint20": Object {
            "RGB": "236,236,236",
            "hex": "#ECECEC",
            "on": "#272727",
          },
          "tint40": Object {
            "RGB": "199,199,199",
            "hex": "#C7C7C7",
            "on": "#272727",
          },
          "tint60": Object {
            "RGB": "160,160,160",
            "hex": "#A0A0A0",
            "on": "#FFFFFF",
          },
          "tint80": Object {
            "RGB": "105,105,105",
            "hex": "#696969",
            "on": "#FFFFFF",
          },
        },
        "complimentary": Object {
          "dark": Object {
            "RGB": "250,175,64",
            "hex": "#FAAF40",
            "on": "#272727",
          },
          "light": Object {
            "RGB": "249,240,21",
            "hex": "#F9F015",
            "on": "#272727",
          },
          "main": Object {
            "RGB": "253,216,53",
            "hex": "#FDD835",
            "on": "#272727",
          },
        },
        "error": Object {
          "dark": Object {
            "RGB": "143,6,6",
            "hex": "#8F0606",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "220,16,16",
            "hex": "#DC1010",
            "on": "#FFFFFF",
          },
        },
        "primary": Object {
          "dark": Object {
            "RGB": "11,102,54",
            "hex": "#0B6636",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "145,208,62",
            "hex": "#91D03E",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "1,145,68",
            "hex": "#019144",
            "on": "#FFFFFF",
          },
        },
        "secondary": Object {
          "dark": Object {
            "RGB": "72,140,155",
            "hex": "#488C9B",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "162,235,252",
            "hex": "#A2EBFC",
            "on": "#272727",
          },
          "main": Object {
            "RGB": "118,182,196",
            "hex": "#76B6C4",
            "on": "#FFFFFF",
          },
        },
        "surface": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
        "white": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
      }
    }
  >
    <div
      className="c0"
      name="email"
    />
  </StyledComponent>
</StyledErrorMessage>
`;
